src/redux/categories/categorySlice.js

```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categories: [],
+};
+
+const categorySlice = createSlice({
+  name: 'category',
+  initialState,
+  reducers: {
+    checkStatus: () => ({ categories: 'Under construction' }),
+  },
```

Comment:
- [ ] The checkStatus action is expected to take the state as an argument and which is then get used for updating that state. Kindly pass the state into your reducer action 👍🏽

```javascript
checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
```